cmd/etcd-mesos-scheduler/app.go

```diff
@@ -122,12 +124,16 @@ func main() {
 		}
 	}
 
+	if *advertiseAddress == "" {
+		*advertiseAddress = *address
```

Comment:
I pretty much copied the `*address` default assignment logic, which felt a little hacky also :/ This simply duplicates the current logic though